Eliran/update docs

thehobbit85 opened the pull request:
Remove some old docs, add some new ones, and prettify the rest :)

thehobbit85 committed:
Remove old irrelevant docs

paullinator commented:
I’d like a sidebar on this. Many of the removed docs are still relevant

samholmes commented on scripts/generateToc.js:
```diff
@@ -0,0 +1,116 @@
+    if (file === TOC_FILE_NAME) continue
+
+    const markdownFile = fs.readFileSync(path.join(startPath, `${file}`))
+    const title = markdownFile.toString().split('\n')[0].split(' &nbsp; ')[1]
```
Why are you taking the first line of the file and the second part of the line after " &nbsp; "?

samholmes commented on server.md:
```diff
@@ -127,4 +148,6 @@ import { asApiGetRequest, asApiGetResponse } from 'my-server'
 
-Because server versions track the API changes of the HTTP interface and _not_ the versioning of any exported types, there are no guarantees that types wont have breaking changes from one server version to another. This is one of the caveats to exporting types from the server, and why it is recommended to define and export types on a complementary client-side library for the server (see 'Client Li ...
\ No newline at end of file
+Because server versions track the API changes of the HTTP interface and _not_ the versioning of any exported types, there are no guarantees that types wont have breaking changes from one server version to another. This is one of the caveats to exporting types from the server, and why it is recommended to define and export types on a complementary client-side library for the server (see 'Client Li ...
```
Missing the period.

thehobbit85 replied:
You are right

samholmes commented on docs.md:
```diff
@@ -0,0 +1,54 @@
+
+- content - The actual content of the page which should only use "native" markdown syntax, without any html tags.
+
+[Back to the top](#developer-documentation-guidelines)
```
Shouldn't this be `#conventions-documentation-guidelines`?

thehobbit85 replied:
Yes it should

samholmes commented on docs.md:
````diff
@@ -0,0 +1,54 @@
+
+```markdown
+
+# [<](README.md) &nbsp; ${main_header}
````
It doesn't seem like the back-to-top links work unless you format the main_header_dashed like so `--conventions-documentation-guidelines`. This is because of the `" &nbsp; "` formatting.

How about:

```md
# ${main_header}
[<- BACK](README.md)

${table_of_content}

&nbsp;

${content}

[Back to the top](#${main_header_dashed})
```

Doing away with the back in the header and just making it its own line?

thehobbit85 replied:
I really like it in the same line so I'm just adding `--` everywhere

samholmes commented on docs.md:
```diff
@@ -0,0 +1,54 @@
+For example for this page it's: "Developer Documentation Guidelines")
+
+- main_header_dashed - the same as the "main_header" only all lower caps and separated by dashes instead of spaces.\
+For example for this page it's: "developer-documentation-guidelines")
```
If we're keeping the back link in the header, then we need to explain to add a `--` prefix to the main_header_dashed.

thehobbit85 replied:
It seems to be working even without it.
But if you insist then I can add it and also fix it in all the pages.

thehobbit85 replied:
I just added it in the template

samholmes reviewed (commented):
Misspell "extentions" for JSON files in `tools/vscode/recommended/`

samholmes reviewed (changes requested):
Requested some changes. Also, can't speak to the removal of some files as mentioned by @paullinator. Perhaps, further discussion needed?

thehobbit85 commented:
@samholmes  I fixed everything you asked and some other small things I found.
Let's schedule the discussion you wanted.

samholmes commented on coding.md:
```diff
@@ -1,32 +0,0 @@
-
-// correct
-const denomination = {
-  currencyCode: 'USD' // Todo: Replace hard coded currencies with library -paulvp
```
Keep this and put it in "Comment Conventions" section

samholmes commented on naming.md:
```diff
@@ -1,106 +0,0 @@
-# Naming Conventions
```
ESLint (standard kit) takes care of most of these conventions.

samholmes commented on naming.md:
````diff
@@ -1,106 +0,0 @@
-
-* Postfix the names of index files with the name of its parent directory
-
-```javascript
-// incorrect
-src/
-  components/
-    MyComponent/
-      index.js // <--
-      MyComponent.js
-      styles.js
-
-// correct
-src/
-  components/
-    MyComponent/
-      indexMyComponent.js // <--
-      MyComponent.js
-      styles.js
-```
````
Arguable whether it's worth keeping this convention.

samholmes commented on naming.md:
````diff
@@ -1,106 +0,0 @@
-const edgeTransactions: Array<EdgeTransaction> = getTransactions()
-```
-
-* Only use named exports
- ```javascript
-// incorrect
-export default class FullWalletListRow extends Component<OwnProps> {
-
- // correct
-export class FullWalletListRow extends Component<OwnProps> {
-```
````
Worth keeping this one around.

samholmes commented on react.md:
```diff
@@ -1,73 +0,0 @@
-# React Conventions
```
This should be kept but updated.

samholmes commented on redux.md:
````diff
@@ -1,110 +0,0 @@
-
-### Async Error Handling
-
-```javascript
-import { InsufficientFundsError } from ‘edge-core-js’
-
-WALLET_API.someCall(edgeWallet, someInputs).then(handleSuccess, handleError)
-
-const handleSuccess = (someData) => {...do stuff with data}
-
-const handleError = (error: Error) => {
-  if (error instanceof InsufficientFundsError) {
-    return dispatch(displayInsufficientFundsAlert())
-  }
-  dispatch(unknownError(error)) // Handles all other error types
-}
-```
-
-### Connectors
-* Do not create new objects in connectors
-
-```javascript
-const defaults = {
-  d: 123,
-  e: 456
-}
-
-const mapStateToProps = (state: State) => {
-  return {
-    a: {
-      ...defaults,
-      ...state.a
-    },
-    b: state.b,
-    c: state.c
-  }
-}
-
-export const connectedMyComponent = connect(mapStateToProps, mapDispatchToProps)(MyComponent)
-```
````
Keep these

samholmes commented on testing.md:
```diff
@@ -1,348 +0,0 @@
-# Testing Conventions
```
These conventions were never reinforced. Argument is to not keep it; perhaps redefine our testing conventions.

samholmes commented on workflow.md:
```diff
@@ -1,36 +0,0 @@
-# Workflow Conventions
```
Decided not to keep

samholmes reviewed (changes requested):
A few changes needed after reviewing deleted files.

thehobbit85 committed:
Added a script to generate the table of content and all the build files/setup files needed

thehobbit85 committed:
Git Conventions

1. Moved the git conventions into their own subsection.
2. Splitted the "git.md" into the PR and commit sections.
3. Added content tables, both for the folder, and for the individual files.
4. Other markdown cleanups

thehobbit85 committed:
Code Conventions

thehobbit85 committed:
Utilities Conventions

1. libs.md - A list of libraries written by Edge.
2. apps.md - Recommended development apps and configuration.

thehobbit85 committed:
Added the new main README and the Documentation Conventions file
